Inputs.md

```diff
@@ -0,0 +1,102 @@
+This guide will describe how inputs from the user (keyboard, mouse, etc.) are routed through InputWritr and how the DeviceLayr Gamepad can be used on top of it.
+
+### Table of Contents
+1. [InputWritr](#inputwritr)
+    1. [Events and Triggers](#events-and-triggers)
+    2. [Running Events](#running-events)
+2. [DeviceLayr](#devicelayr)
+
+# InputWritr
+
+## Events and Triggers
+
+InputWritr is a module that automates interactions with user inputs and events. 
+
+Inputs are stored as both character codes (e.g. 37) and as a string representation of themselves (e.g. "left").
+Triggers are what tie inputs with events and are specific actions on the inputs (e.g. press).
+
+Events can be added using `addEvent` which takes in a trigger name, key code, and the function to run when the event is triggered.
+
+```typescript
+InputWritr.addEvent("onKeyDown", 37, () => { console.log("left button pressed"); });
+```
+
+...and are removed with `removeEvent` which takes in the trigger and key code.
+
+```typescript
+InputWritr.removeEvent("onKeyDown", 37);
+```
+
+Events can be triggered by any number of inputs.
+
+Aliases are additional inputs that allow for an event to be triggered from multiple user sources.
+They can be added and removed using `addAliasValues` and `removeAliasValues` which both take in the input and an array of key character codes as arguments.
+
+```typescript
+InputWritr.addAliasValues("left", [65, 74, 49]); // keys a, j, and 1 respectively 
+InputWritr.removeAliasValues("left", [65, 74, 49]);
+```
+
+```typescript
+let InputWriter: IInputWritr = new InputWritr({
+    "aliases": {
+        "left":   [65, 37], // a, left button
+    },
+    "triggers": {
+        "onkeydown": {
+            "left": () => { console.log("left button was pressed"); }
+        }
+    }
+});
+```
+
+## Running Events
+
+To run events, use `callEvent` and `makePipe`.
+`makePipe`  takes in a trigger, code label to get the alias from the event, and a boolean to determine whether the default event should be prevented and
+returns a function to run the triggered event.
+
+```typescript
+InputWritr.makePipe("onkeydown", "left", true);
```

Comment:
You should add an example of this being used; otherwise it's not clear what the usage is.